src/options/getPackageOptions.ts

```diff
@@ -26,11 +27,17 @@ export function getCombinedPackageOptions(actualPackageOptions: Partial<PackageO
  */
 export function getPackageOptions(packagePath: string): Partial<PackageOptions> {
   const configExplorer = cosmiconfigSync('beachball', { cache: false });
+  console.log(configExplorer);
   try {
     const results = configExplorer.load(path.join(packagePath, 'package.json'));
-    return (results && results.config) || {};
+    if (results && results.config) {
```

Comment:
Hmm, as far as logging the error is concerned, on many an occasion I found it less useful than an explicit formatted error message like here. As far as readability goes, I think it' ok as it is now, however, your suggestion does eliminate throwing an empty `Error` just to catch in a block below. I don't have any real measurements, but usually exception handling is pretty costly, so this might be a better way indeed.

Reply:
We could also replace

```ts
return (results && results.config) || {};
```

with

```ts
return results?.config || {};
```

which a lot more readable IMHO. (Thanks @andrefcdias for suggestion!)

Reply:
I quite like @Hotell 's suggestion, since there's a clear differentiation between `none` and `empty`

I can turn the expression to use null propagators, but it's just syntactic sugar and the `&&` is still much more prevalent in the JS world

Reply:
**thought:** short circuiting might be dangerous in some cases, nor is it explicit (readable?) -> thus if one really wants to use "shorthands" I recommend use ternary

Reply:
> I don't have any real measurements, but usually exception handling is pretty costly, so this might be a better way indeed.

@kubkon Practically, cost of exception handling isn't a concern in a situation like this. The operation isn't run frequently enough or in a particularly performance-critical situation for it to matter.